tests/advanced_visibility_test.go

```diff
@@ -540,59 +540,87 @@ func (s *advancedVisibilitySuite) TestListWorkflow_KeywordQuery() {
 	s.NoError(err)
 	s.Len(resp.GetExecutions(), 0)
 
-	if s.isElasticsearchEnabled {
-		// LIKE is supported on Elasticsearch only.
+	// Prefix search
+	listRequest = &workflowservice.ListWorkflowExecutionsRequest{
+		Namespace: s.namespace,
+		PageSize:  defaultPageSize,
+		Query:     `CustomKeywordField STARTS_WITH "justice"`,
+	}
+	resp, err = s.engine.ListWorkflowExecutions(NewContext(), listRequest)
+	s.NoError(err)
+	s.Len(resp.GetExecutions(), 1)
```

Comment:
Can you validate that you retrieve the correct record, not that we retrieve _a_ record? We could be retrieving the wrong record, but this test would still pass.

As a reviewer I'm not confident in this test

Reply:
@tdeebswihart This test is new.